gradle-plugin/src/main/kotlin/io/spine/protodata/gradle/plugin/Extension.kt

```diff
@@ -81,12 +83,64 @@ public class Extension(internal val project: Project): CodegenSettings {
     internal fun requestFile(forSourceSet: SourceSet): Provider<RegularFile> =
         requestFilesDirProperty.file(CodeGeneratorRequestFile.name(forSourceSet))
 
+    override val paths: Multimap<String, SourcePaths> = HashMultimap.create()
+
+    /**
+     * Obtains the source configured paths.
+     *
+     * If the deprecated [subDirs] and [targetBaseDir] are used, constructs [SourcePaths] instances
+     * from the present data for backward compatibility. However, in this compatibility mode,
+     * only Java and Kotlin source file sets can be constructed.
+     */
+    internal fun pathsOrCompat(sourceSet: SourceSet): Set<SourcePaths> {
+        if (!paths.isEmpty) {
+            return paths[sourceSet.name].toSet()
+        }
+        val srcRoots = sourceDirs(sourceSet).get()
+        val targetRoots = targetDirs(sourceSet).get()
+        return srcRoots.zip(targetRoots)
+            .map { (src, target) ->
+                val pathSuffix = src.asFile.toPath().name
+                val lang = if (pathSuffix == "kotlin") Kotlin else Java
```

Comment:
Instead of using a literal, maybe we could use `knownLanguages`, which we have previously created exactly to associate `Kotlin` with `"kotlin"`?